images/base/files/usr/local/bin/entrypoint

```diff
@@ -416,10 +437,18 @@ enable_network_magic(){
         echo "ERROR: Have an old IPv4 address but no current IPv4 address (!)" >&2
         exit 1
       fi
-      # kubernetes manifests are only present on control-plane nodes
-      sed -i "s#${old_ipv4}#${curr_ipv4}#" /etc/kubernetes/manifests/*.yaml || true
-      # this is no longer required with autodiscovery
-      sed -i "s#${old_ipv4}#${curr_ipv4}#" /var/lib/kubelet/kubeadm-flags.env || true
+      if [[ "${old_ipv4}" != "${curr_ipv4}" ]]; then
+        # kubernetes manifests are only present on control-plane nodes
+        sed_ipv4_command="s#$(regex_escape_ip "${old_ipv4}")#${curr_ipv4}#g"
+        sed -i "${sed_ipv4_command}" /etc/kubernetes/manifests/*.yaml || true
+        sed -i "${sed_ipv4_command}" /etc/kubernetes/controller-manager.conf || true
+        sed -i "${sed_ipv4_command}" /etc/kubernetes/scheduler.conf || true
+        sed -i "${sed_ipv4_command}" /kind/kubeadm.conf || true
+        # this is no longer required with autodiscovery
+        sed -i "${sed_ipv4_command}" /var/lib/kubelet/kubeadm-flags.env || true
+        # certificate must match the new IP
+        fix_certificate || true
```

Comment:
lets store above `local should_fix_certificate=false` and then in both the ipv4 and ipv6 paths do `should_fix_certificate=true` here.

then run this only once below.

Reply:
also, do we need to swallow the exit status of the function?
should not better do that inside and not have to do it here?

Reply:
i think that's just matching the existing pattern for ip change above.

we swallow the error replacing file contents currently. we probably should not, we should check for files and error if trying to replace in them fails.

Reply:
updated to check file existence first and not swallow error